File: RetrofitAPI/app/src/main/java/com/example/retrofitapi/MainActivity.java
package com.example.retrofitapi;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    private Button getPhotoActivity;
    private Button getPhotoById;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        getPhotoActivity = findViewById(R.id.getPhotos);
        getPhotoById = findViewById(R.id.getId);

        getPhotoActivity.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, MainActivity_getPhoto.class);
                startActivity(intent);
            }
        });

        getPhotoById.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, MainActivity_getPhoto.class).putExtra("byId", true);
                startActivity(intent);
            }
        });
    }
}

File: AulaActivityFragment/app/src/main/java/com/example/aulaactivityfragment/MainActivity.kt
package com.example.aulaactivityfragment

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button

class MainActivity : AppCompatActivity() {

    private lateinit var buttonAbrir: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.i("ciclo_vida", "onCreate")
        setContentView(R.layout.activity_main)

        buttonAbrir = findViewById(R.id.button_abrir)
        buttonAbrir.setOnClickListener {
            val intent = Intent(this, DetalhesActivity::class.java)

            // passando dados para outra activity
//            intent.putExtra("filme", "The Witcher")
//            intent.putExtra("classificacao", 5)
//            intent.putExtra("avaliacoes", 9.2)

            val filme = Filme(
                "Sem Limites",
                "teste",
                4.5,
                "Pedro",
                "Netflix"
            )

            intent.putExtra("filme", filme)

            startActivity(intent)
        }
    }

    override fun onStart() {
        super.onStart()

        // Simulando carregamento de dados
        Log.i("ciclo_vida", "onStart")
    }

    override fun onResume() {
        super.onResume()
        Log.i("ciclo_vida", "onResume")
    }

    override fun onPause() {
        super.onPause()
        Log.i("ciclo_vida", "onPause")
    }

    override fun onStop() {
        super.onStop()
        Log.i("ciclo_vida", "onStop")
    }

    override fun onRestart() {
        super.onRestart()
        Log.i("ciclo_vida", "onRestart")
    }

    override fun onDestroy() {
        Log.i("ciclo_vida", "onDestroy")
        super.onDestroy()
    }

}